src/Controls/src/Core/Shell/ShellNavigationManager.cs

```diff
@@ -309,7 +309,16 @@ public static void ApplyQueryAttributes(Element element, ShellRouteParameters qu
 
 
 			if (baseShellItem is ShellContent)
-				baseShellItem.ApplyQueryAttributes(MergeData(element, filteredQuery, isPopping));
+			{
+				var mergedData = MergeData(element, filteredQuery, isPopping);
+
+				//if we are pop or navigating back, we need to apply the query attributes to the ShellContent
+				if(isPopping)
```

Comment:
nit: white space

Reply:
I have addressed the whitespace issue. Let me know if anything else needs adjustment.